### PR TITLE
Release 16.4.0

### PR DESCRIPTION
**Pull request by r0mant**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,51 @@
 # Changelog
 
+## 16.4.0 (09/18/2024)
+
+### Machine ID for Terraform Cloud
```
Comment (zmb3) [inline CHANGELOG.md:5]: @timothyb89 should we say `Terraform Enterprise` instead?

Can we add the docs link here too?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,51 @@
...
+trust domains managed by other SPIFFE compatible platforms. Establishing a
+relationship between the trust domains enables workloads belonging to one trust
+domain to validate the identity of workloads in the other trust domain, and vice
+versa.
+
+### Multi-domain support for web applications
+
+Teleport now supports web application access where one application depends on
+another. For example, you may have a web application that depends on a backend
+API service, of which are separate apps protected by Teleport.
```
Comment (zmb3) [inline CHANGELOG.md:23]:
```suggestion
API service, both of which are separate apps protected by Teleport.
```

**Review by zmb3 — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (timothyb89) [reply CHANGELOG.md:5]: Ah, hmm. I think officially the name is HCP Terraform, and Terraform Enterprise is the separate self hosted version of that, which we also support. They renamed it from Terraform Cloud earlier this year.

The v16 docs backport is in the merge queue. They're available at https://goteleport.com/docs/ver/17.x/admin-guides/infrastructure-as-code/terraform-provider/terraform-cloud/ for the moment.

**Review by tigrato — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (zmb3) [reply CHANGELOG.md]: That is a really awkward limitation that we should fix. The changelog is one of the most important places to have docs links!

cc @xinding33 @ptgott can we make this a docs team goal for Q4?

Comment (r0mant) [reply CHANGELOG.md]: Yep, agreed.

**Pull request merged**